presto-main/src/test/java/com/facebook/presto/block/BlockAssertions.java

```diff
@@ -519,4 +595,183 @@ public static RunLengthEncodedBlock createRLEBlock(long value, int positionCount
+        return new DictionaryBlock(dictionary, ids);
+    }
+
+    public static Block createDictRleBlock(Block block, int positionCount, List<Encoding> wrappings)
+    {
+        checkArgument(!wrappings.isEmpty());
+
+        Block wrappedBlock = block;
+
+        for (int i = wrappings.size() - 1; i >= 0; i--) {
+            switch (wrappings.get(i)) {
+                case DICTIONARY:
+                    wrappedBlock = createDictionaryBlock(wrappedBlock, positionCount);
+                    break;
+                case RUNLENGTH:
+                    wrappedBlock = createRLEBlock(wrappedBlock, positionCount);
+                    break;
+                default:
+                    throw new IllegalArgumentException(format("wrappings %s is incorrect", wrappings));
+            }
+        }
+        return wrappedBlock;
+    }
+
+    public static Block createBlockWithType(Type type, int positionCount, boolean allowNulls, boolean isView, List<Encoding> wrappings)
+    {
+        Block block = null;
+
+        if (isView) {
+            positionCount *= 2;
+        }
+
+        if (type == BOOLEAN) {
+            block = createBooleanBlock(positionCount, allowNulls);
+        }
+        else if (type == BIGINT) {
+            block = createLongsBlock(positionCount, allowNulls);
+        }
+        else if (type == INTEGER) {
+            block = createIntsBlock(positionCount, allowNulls);
+        }
+        else if (type == SMALLINT) {
+            block = createSmallintBlock(positionCount, allowNulls);
+        }
+        else if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            if (decimalType.isShort()) {
+                block = createLongsBlock(positionCount, allowNulls);
+            }
+            else {
+                block = createLongDecimalsBlock(positionCount, allowNulls);
+            }
+        }
+        else if (type == VARCHAR) {
+            block = createStringsBlock(positionCount, allowNulls, BlockAssertions.MAX_STRING_SIZE);
+        }
+        else {
+            // Nested types
+            // Build isNull and offsets of size positionCount
+            boolean[] isNull = new boolean[positionCount];
+            int[] offsets = new int[positionCount + 1];
+            for (int position = 0; position < positionCount; position++) {
+                if (allowNulls && position % 7 == 0) {
+                    isNull[position] = true;
+                    offsets[position + 1] = offsets[position];
+                }
+                else {
+                    offsets[position + 1] = offsets[position] + (type instanceof RowType ? 1 : BlockAssertions.RANDOM.nextInt(BlockAssertions.ENTRY_SIZE) + 1);
+                }
+            }
+
+            // Build the nested block of size offsets[positionCount].
+            if (type instanceof ArrayType) {
+                Block valuesBlock = createBlockWithType(
```

Comment:
this method is quite long; I'd put all arguments on the same line to save some vertical space

`Block valuesBlock = createBlockWithType(((ArrayType) type).getElementType(), offsets[positionCount], allowNulls, isView, wrappings);`